### PR TITLE
Update GitHub issue templates

### LINKED ISSUE #1366: Require must-gather in bug reports
When `must-gather` lands and we deem it as working good enough, we should require it for bug reports and update the issue templates.

Probably also update the README.

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR moves the existing GitHub issue templates to structured YAML and restructures them, now that we have `must-gather`.

**Which issue is resolved by this Pull Request:**
Resolves #1366

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Bug Report
+description: Report a bug encountered with Scylla Operator
+labels: kind/bug
+body:
+- type: textarea
+  id: problem
+  attributes:
+    label: What happened?
+    description: |
+      Please provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner.
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:10]: I'd reword it to sound more like an encouragement than a threat. ;) How about "Please provide as much information as possible. Ensuring all details are shared will help us address your bug in a timely manner."

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  id: expected
+  attributes:
+    label: What did you expect to happen?
+  validations:
+    required: true
+
+- type: textarea
+  id: repro
+  attributes:
+    label: How can we reproduce it (as minimally and precisely as possible)?
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:24]:
```suggestion
   label: How can we reproduce it? Provide the fewest and most precise steps possible.
```
nit

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      Kubernetes platform info:
+      </details>
+  validations:
+    required: true
+
+- type: textarea
+  id: must-gather
+  attributes:
+    label: |
+      Please attach must-gather archive. (Must-gather archive is required for every bug report.)
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:55]:
```suggestion
      Please attach the must-gather archive. The must-gather archive is a **mandatory** part of every bug report.
```
nit

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      See https://operator.docs.scylladb.com/stable/support/must-gather.html to learn how you can collect it.
+      
+      Tip: You can attach images or log files by clicking this area to highlight it and then dragging files in.
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:58]:
```suggestion
      Tip: You can attach images or log files by clicking this area to highlight it and then dragging the files in.
```

File: .github/ISSUE_TEMPLATE/ci-flake.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  attributes:
+    label: Link to the job that flaked.
+  validations:
+    required: true
+
+- type: textarea
+  id: details
+  attributes:
+    label: Snippet of what failed.
+    description: Inset a short snippet of what failed to have a quick preview (stack trace) and something to track even if logs would be gone.
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/ci-flake.yaml:16]:
```suggestion
    description: Insert a short snippet of what failed (stack trace) to provide a quick preview even if the logs are missing.
```

File: .github/ISSUE_TEMPLATE/feature-request.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  id: expected
+  attributes:
+    label: What should the feature do?
+  validations:
+    required: true
+
+- type: textarea
+  id: expected
+  attributes:
+    label: What is use case behind this feature?
```
Comment (rzetelskik) [inline .github/ISSUE_TEMPLATE/feature-request.yaml:15]:
```suggestion
    label: What is the use case behind this feature?
```

**Review by rzetelskik — COMMENTED** [review]
is there a way to see what it looks like after parsing?

**Comment by tnozicka** [comment]
> is there a way to see what it looks like after parsing?

i think you can test it in settings but not now because we are transitioning from the md files

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 2 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/bug-report.yaml:24]: it's identical to the kube issue tracker

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 4 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/bug-report.yaml:58]: since these aren't specific files, I am not sure `the` belongs there. Also this is taken from GH docs where it is identical https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 1 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/bug-report.yaml]: also identical to Kube but rephrased

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 3 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/bug-report.yaml]: done

File: .github/ISSUE_TEMPLATE/feature-request.yaml (hunk 6 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/feature-request.yaml]: done

File: .github/ISSUE_TEMPLATE/ci-flake.yaml (hunk 5 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/ci-flake.yaml]: fixed the typo but the new wording doesn't retain causality

Comment (rzetelskik) [reply .github/ISSUE_TEMPLATE/ci-flake.yaml]: let's leave the wording then but the `would be gone` part is ungrammatical, imo either are or were

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 2 above)
Comment (rzetelskik) [reply .github/ISSUE_TEMPLATE/bug-report.yaml:24]: I think overusing the parentheses obfuscates the text, but I don't feel strong about it. On a side note, I don't see it being identical to the wording in kube repo as a convincing argument - in terms of code, sure, but stuff like this doesn't have to be some golden standard there.

File: .github/ISSUE_TEMPLATE/ci-flake.yaml (hunk 5 above)
Comment (tnozicka) [reply .github/ISSUE_TEMPLATE/ci-flake.yaml]: ok, I can settle for `are`

**Comment by rzetelskik** [comment]
/lgtm
thanks
